### PR TITLE
Remove NuGetProvider submodule, update build script to use NuGetProvider NuGet package

### PR DESCRIPTION
**Pull request by brywang-msft**
This should finish the separation of NuGetProvider and OneGet. Pulls the NuGetProvider .dll from 'https://powershell.myget.org/feed/Packages/powershellmodule' instead of building it here.

### CODE REVIEW

File: src/build.ps1 [hunk 1]
```diff
@@ -67,6 +67,16 @@ if ($Framework -eq "all")
     $frameworks = @($Framework)
 }
 
+$providersFromPackages = @(
+    @{
+        'Name' = 'Microsoft.PowerShell.PackageManagement.NuGetProvider'
+        'ProviderName' = 'NuGet'
+        #'Location' = 'https://powershell.myget.org/F/powershellmodule/api/v2'
+        'Location' = 'E:\Packages_bk\PackagesTesting'
```
Comment (edyoung) [inline src/build.ps1]: will this location work for others?

File: src/build.ps1 [hunk 2]
```diff
@@ -179,4 +224,8 @@ foreach ($currentFramework in $frameworks)
     Add-Type -assemblyname System.IO.Compression.FileSystem
     Write-Verbose "Zipping $sourcePath into $packageFileName" -verbose
     [System.IO.Compression.ZipFile]::CreateFromDirectory($sourcePath, $packageFileName) 
+}
+
```
Comment (edyoung) [inline src/build.ps1]: should this be in a finally block?

File: src/build.ps1 [hunk 3]
```diff
@@ -179,4 +224,8 @@ foreach ($currentFramework in $frameworks)
...
+foreach ($unregisterPackageSource in $unregisterPackageSources) {
+    $unregisterPackageSource | Unregister-PackageSource
```
Comment (edyoung) [inline src/build.ps1]: wouldn't $unregisterPackagesSources | Unregister-PackageSource suffice?

File: src/pack.ps1 [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+        $psd1 = Get-ChildItem -Path (Join-Path -Path $OneGetModulePath -ChildPath "PackageManagement.psd1")
+        if ($psd1) {
+            $version = (Test-ModuleManifest -Path $psd1).Version.ToString()
+        } else {
+            Write-Warning "Couldn't find psd1 file in root of '$OneGetModulePath'. Is this the OneGet module directory you want to pack?"
+        }
+    }
+
+    if (-not $version) {
+        throw "Couldn't discover version of package - is Microsoft.PackageManagement.dll there?"
```
Comment (edyoung) [inline src/pack.ps1]: include $version?

File: src/build.ps1 [hunk 5]
```diff
@@ -165,6 +173,43 @@ foreach ($currentFramework in $frameworks)
     CopyToDestinationDir $dscResourceItemsPackage (Join-Path -Path $destinationDirDscResourcesBase -ChildPath "MSFT_PackageManagement")
     CopyToDestinationDir $dscResourceItemsPackageSource (Join-Path -Path $destinationDirDscResourcesBase -ChildPath "MSFT_PackageManagementSource")
 
+    # Download and copy providers from packages
+    foreach ($providerPackageInfo in $providersFromPackages) {
+        $packageSource = Get-PackageSource -Location $providerPackageInfo.Location -ProviderName $providerPackageInfo.ProviderName
+        if (-not $packageSource) {
+            $packageSource = Register-PackageSource -Name ([Guid]::NewGuid().ToString()) -Location $providerPackageInfo.Location -ProviderName $providerPackageInfo.ProviderName -Trusted
```
Comment (jianyunt) [inline src/build.ps1]: Could you please add comments about the source location here? it is from  'Location' = 'E:\Packages_bk\PackagesTesting' ? How the package installed to this location?

Comment (brywang-msft) [reply src/build.ps1]: I think you're looking at an old commit - see the latest: 
ada1ea8

**Comment by brywang-msft** [comment]
Looking into: https://ci.appveyor.com/project/jianyunt/oneget/build/1.0.502

File: src/build.ps1 (hunk 5 above)
Comment (jianyunt) [reply src/build.ps1]: ok

**Comment by jzabroski** [comment]
> This should finish the separation of NuGetProvider and OneGet. Pulls the NuGetProvider .dll from the powershellmodule NuGet feed instead of building it here.

@brywang-msft 
What is the "powershellmodule NuGet feed"?

**Comment by brywang-msft** [comment]
@jzabroski https://powershell.myget.org/feed/Packages/powershellmodule - updated initial comment as well

**Comment by jianyunt** [comment]
@jzabroski "powershellmodule NuGet feed" is our internal nuget feed while building the code we publish the package to the site.

**Comment by jzabroski** [comment]
I'm so ridiculously confused by how Microsoft learned how to manage packages. I've been trying to reach out to Microsoft for years and explain why they over complicate things, but no luck.

Can I ask why you are injecting a hard dependency on myget?

This really, really is not how distributed package management is supposed to work. You guys really need to study things like Thomas Lord's Arc (tla) and the Debian Package Manager to truly understand distributed computing. It's a bit frustrating since NuGet is what ships in Visual Studio despite Paket being superior.

**Comment by jzabroski** [comment]
@jianyunt This is GitHub.  There is no internal feeds.

**Comment by jzabroski** [comment]
If you want to add a reference, embed a checksum.

**Comment by jianyunt** [comment]
@jzabroski  You are right. This is the public github, no internal.  We should not assume build.ps1 is only used by the appveryor for the nightly or checkin build. @brywang-msft, @edyoung  let's rethink about it. Thanks.

**Comment by brywang-msft** [comment]
Thanks for the input everyone. I will be closing this PR without merging until further discussion can be had. Thanks again for your time.

**Pull request closed without merging**